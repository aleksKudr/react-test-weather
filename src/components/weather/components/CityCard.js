import React from 'react';

class CityCard extends React.PureComponent {
  render() {
    // console.log('this.props', JSON.stringify(this.props));
    if (this.props.name.main === undefined && this.props.weather === undefined) {
      return <div />;
    } return (


      <div className="row">
        <div className="col">
          {this.props.name.dt_txt}
        </div>
        <div className="col">
          {(((1.8 * (this.props.name.main.temp - 273) + 32) - 32) / 1.8).toFixed(2)}
        </div>
        <div className="col">
          {this.props.name.weather[0].main}
        </div>
      </div>
    );
  }
}

export default CityCard;
