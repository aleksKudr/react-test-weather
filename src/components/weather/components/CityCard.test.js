import React from 'react';
import CityCard from './CityCard';

const props = { name: { dt: 1546322400,
  main: { temp: 263.875,
    temp_min: 263.875,
    temp_max: 263.875,
    pressure: 1023.04,
    sea_level: 1031.42,
    grnd_level: 1023.04,
    humidity: 76,
    temp_kf: 0 },
  weather: [{ id: 600,
    main: 'Snow',
    description: 'light snow',
    icon: '13d' }],
  clouds: { all: 32 },
  wind: { speed: 1.71,
    deg: 234.501 },
  snow: { '3h': 0.069999999999999 },
  sys: { pod: 'd' },
  dt_txt: '2019-01-01 06:00:00' } };

const component = shallow(
  <CityCard {...props} />
);

it('test snapshot', () => {
  expect(component).toMatchSnapshot();
});
