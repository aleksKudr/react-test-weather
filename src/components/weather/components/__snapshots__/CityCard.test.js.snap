// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CityCard
    name={
      Object {
        "clouds": Object {
          "all": 32,
        },
        "dt": 1546322400,
        "dt_txt": "2019-01-01 06:00:00",
        "main": Object {
          "grnd_level": 1023.04,
          "humidity": 76,
          "pressure": 1023.04,
          "sea_level": 1031.42,
          "temp": 263.875,
          "temp_kf": 0,
          "temp_max": 263.875,
          "temp_min": 263.875,
        },
        "snow": Object {
          "3h": 0.069999999999999,
        },
        "sys": Object {
          "pod": "d",
        },
        "weather": Array [
          Object {
            "description": "light snow",
            "icon": "13d",
            "id": 600,
            "main": "Snow",
          },
        ],
        "wind": Object {
          "deg": 234.501,
          "speed": 1.71,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col"
        >
          2019-01-01 06:00:00
        </div>,
        <div
          className="col"
        >
          -9.13
        </div>,
        <div
          className="col"
        >
          Snow
        </div>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2019-01-01 06:00:00",
          "className": "col",
        },
        "ref": null,
        "rendered": "2019-01-01 06:00:00",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "-9.13",
          "className": "col",
        },
        "ref": null,
        "rendered": "-9.13",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Snow",
          "className": "col",
        },
        "ref": null,
        "rendered": "Snow",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col"
          >
            2019-01-01 06:00:00
          </div>,
          <div
            className="col"
          >
            -9.13
          </div>,
          <div
            className="col"
          >
            Snow
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2019-01-01 06:00:00",
            "className": "col",
          },
          "ref": null,
          "rendered": "2019-01-01 06:00:00",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-9.13",
            "className": "col",
          },
          "ref": null,
          "rendered": "-9.13",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Snow",
            "className": "col",
          },
          "ref": null,
          "rendered": "Snow",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
